scipy/signal/signaltools.py

```diff
@@ -100,6 +100,19 @@ def correlate(in1, in2, mode='full'):
         ``same``
            The output is the same size as `in1`, centered
            with respect to the 'full' output.
+    method : str {'direct', 'auto', 'fft'}, optional
+        A string indicating what method used to perform the convolution.
+
+        ``direct``
+           The convolution is determined directly from sums, the definition of
+           convolution.
+        ``fft``
+           The Fourier Transform is used to perform the correlation and only
+           works for numerical arrays.
+        ``auto``
```

Comment:
In this list, I think it makes sense to leave `auto` last, though, since understanding it depends on the previous descriptions

Also should say "See Notes" and then in the Notes section, describe the caveats of how it always uses `direct` mode for object arrays or long ints, etc.